Add TACo landing page (#113)
threshold-network/website: src/content/pages/build/taco.md

---
template: build-page/taco
path: /build/taco
title: TACo Page
seoTitle: Threshold Taco
seoDescription: "TACo (Threshold Access Control) enables end-to-end encrypted data sharing and communication."
tacoInfo:

>   title: "TACo: Privacy without trust"

Reviewer: This should be "TACo: Privacy Without Trust" to be consistent with the casing on similar pages (eg tBTC SDK)

Author: Addressed in [capitalize the taco info title](https://github.com/threshold-network/website/pull/113/commits/5d9ddaef765b53c13853149e099639572ae97a90)

>   description: "Give your users true sovereignty over their data, today. TACo – the first and only access control layer you aren’t forced to trust."

Reviewer 2: ```suggestion
  description: "Give your users true sovereignty over their data, today. TACo is the first and only access control layer you aren’t forced to trust."
```

  image: /images/taco-encryption.svg
  buttons:
    - label: Start Building
      url: https://docs.threshold.network/app-development/threshold-access-control-tac/integration-guide
      variant: EXTERNAL_SOLID
    - label: Trust Overview

>       url: https://docs.threshold.network/app-development/threshold-access-control-tac/trust-assumptions/cbd-mainnet-version

Reviewer 3: Modified this slug in the docs, please update the link:

```suggestion
      url: https://docs.threshold.network/app-development/threshold-access-control-tac/trust-assumptions/mainnet-trust-model-foundation
```

      variant: EXTERNAL_OUTLINE
  rowReverse: false
tacoFeaturesInfo:

>   title: "The Fully Decentralized End-to-End Encryption"

Reviewer 2: ```suggestion
  title: "End-to-End Encryption that's End-to-End Decentralized"
```
Let's stick with all the copy I wrote in the Figma as the starting point. If I agree with a suggestion, I'll edit the text here.

>   title: "The Fully Decentralized End-to-End Encryption"

Reviewer 2: Yeh that's good! @evandrosaturnino let's use this copy

>   title: "End-to-End Encryption. End-to-End Decentralized."

Reviewer 2: It sounds better when the tense matches, but I think it's more vague. The point here is that the e2ee specifically is what is decentraliz**ed**, as opposed to the system having some form of decentralization.

>   description: "Empower users to share data with qualifying recipients, and no-one else Encryption via a decentralized cryptosystem."

Reviewer 2: ```suggestion
  description: "Empower users to share data with qualifying recipients, and no-one else. Encryption via a decentralized cryptosystem, from day one."
```

>   description: "Empower users to share data with qualifying recipients, and no-one else. Encryption via a decentralized cryptosystem, from day one."

Author: @arjunhassard Any comments on this?

>   description: "Empower users to share data with qualifying recipients, and no-one else. Encryption via a decentralized cryptosystem, from day one."

Reviewer 2: I prefer the first version, it's punchier.

  highlighted: ["Decentralized", "Encryption"]
tacoFeatures:
  - title: "Web3 needs TACo"
    cards:
      - title: "Decentralized Storage"

>         description: "TACo is the perfect compliment to persistent storage, finally enabling users to grant future access without having to download & re-encrypt locally."

Reviewer 2: Nice spot!

        icon: /images/storage-icon.svg
      - title: "Web3 Social Networks"

>         description: "TACo enables many-to-many data sharing, minus a centralized  authority that eventually decides to harvest and monetize interpersonal communication."

Reviewer 2: ```suggestion
        description: "TACo enables many-to-many data sharing, minus a centralized authority that may eventually decide to harvest and monetize interpersonal communication."
```

Reviewer 2: Fine with either! Was loosely referencing the 'Don't Become Evil' component of d/acc

        icon: /images/social-icon.svg
      - title: "Media content-gating"

>         description: "Whether it's a stream, a movie or a collectible, TACo enforces decryption rights to media, so owning an NFT has meaning beyond a symbolic receipt."

Reviewer 2: ```suggestion
        description: "Whether it's a stream, a movie or a collectible, TACo enforces the decryption rights. 'Owning' an NFT now has meaning beyond a symbolic receipt."
```

        icon: /images/media-icon.svg
  - title: "The Web needs TACo"
    cards:
      - title: "Hyper-sensitive data"

>         description: "From reproductive health data to  seed phrase recovery, TACo is leveraged to ensure that high-stakes data sharing doesn’t involve trusting an intermediary."

Reviewer 2: ```suggestion
        description: "From reproductive healthcare to password recovery, TACo is used to avoid trusting an intermediary with high stakes data sharing."
```

Reviewer 4: I liked more the original angle in the last phrase, something like:
```suggestion
        description: "From reproductive healthcare to password recovery, TACo is used to ensure that high-stakes data sharing doesn’t involve trusting an intermediary."
```

Reviewer 2: Yep that works!

        icon: /images/locker-icon.svg
      - title: "IoT & Connected Vehicles"

>         description: "TACo is the antidote to the interconnected hardware panopticon, where every waking (and sleeping) moment of our lives is surveilled."

Reviewer 2: ```suggestion
        description: "TACo is the antidote to the interconnected hardware panopticon, in which every waking (and sleeping) moment of our lives is surveilled/harvested by data custodians."
```

        icon: /images/iot-icon.svg
      - title: "Activism & Human Rights"

>         description: "TACo protects those most at risk of intimidation, censorship and reprisal based on a data leaks – whether via a hack, subpoena, or a data vending business model."

Reviewer 2: ```suggestion
        description: "TACo protects the most vulnerable in society – those who may suffer  intimidation, censorship and violence as a consequence of their data being leaked." 
```

Reviewer 4: 💯

>         description: "TACo protects the most vulnerable in society – those who may suffer  intimidation, censorship and violence as a consequence of their data being leaked."

Reviewer 2: ```suggestion
        description: "TACo protects the most vulnerable in society – those who may face intimidation, censorship and violence as a consequence of their data being leaked."
```

I prefer 'face' over 'suffer'. But 'individuals' here sounds a little odd.

        icon: /images/activism-icon.svg
tacoDiagramsInfo:
  title: "Integrate a Privacy Layer with Nothing to Hide"

>   description: "TACo isn't just auditable, it's comprehensible, from high-level schematic to client code."

Reviewer 2: ![image](https://github.com/threshold-network/website/assets/2332199/fd434fe1-9af8-4463-ac16-31de40af4d26)
 
This subtitle is formatting a bit weirdly. Maybe this small edit will help?

```suggestion
  description: "TACo isn't just auditable, it's comprehensible. From high-level schematic to client code."
```

  image: /images/taco-grid.svg
  background: /images/glowing-background.png
tacoDiagrams:
  - title: "Generating a DKG Cohort"
    preTitle: "learn"

>     description: "Independent nodes sampled from the Threshold network form a cohort – who collectively generate a DKG public key."

Reviewer 2: ```suggestion
    description: "Independent nodes are sampled from the Threshold network form a cohort. They collectively generate a DKG public key."
```

Reviewer 4: "form a cohort" should be corrected. How about something like:
```suggestion
    description: "Independent nodes are sampled from the Threshold Network, forming a cohort that collectively generates a DKG public key."
```

Reviewer 2: _Independent nodes sampled from the Threshold network form a cohort_  is actually correct grammatically. But I think your suggestion is more consistent with the other stages so let's go with that. @evandrosaturnino

>     description: "Independent nodes are sampled from the Threshold Network, forming a cohort that collectively generates a DKG public key."

Reviewer 3: ```suggestion
    description: "Independent nodes are sampled from the Threshold Network to collectively generate a distributed public key."
```

    image: /images/first-taco-diagram.svg
    buttons:
      - label: "nucypher/ferveo"
        url: https://github.com/nucypher/ferveo
        variant: EXTERNAL_OUTLINE
      - label: "nucypher-contracts"
        url: https://github.com/nucypher/nucypher-contracts
        variant: EXTERNAL_OUTLINE
  - title: "Encrypting Private Data"
    preTitle: "learn"

>     description: "Data owners specify conditions for decryption, embedded with private data and encrypted with the DKG public key."

Reviewer 2: ```suggestion
    description: "Data owners specify decryption conditions, embedded alongside the private data. The payload is encrypted with the DKG public key."
```

    image: "/images/second-taco-diagram.svg"
    buttons:
      - label: "taco-web"
        url: https://github.com/nucypher/taco-web
        variant: EXTERNAL_OUTLINE

>       - label: "nucypher/main"
>         url: https://github.com/nucypher/nucypher/tree/main

Reviewer 3: `main` is the default branch for the repos.

```suggestion
      - label: "nucypher"
        url: https://github.com/nucypher/nucypher
```

        variant: EXTERNAL_OUTLINE
  - title: "Accessing Decrypted Data"
    preTitle: "learn"

>     description: "Requesters present state to the cohort, who verify conditions fulfillment before provisioning decryption material."

Reviewer 2: ```suggestion
    description: "Data requesters present proof to the cohort, who verify condition fulfillment before provisioning them with decryption material."
```

>     description: "Data requesters present proof to the cohort, who verify condition fulfillment before provisioning them with decryption material."

Reviewer 3: ```suggestion
    description: "Data requesters present proof to the cohort, who verify condition fulfilment before performing any decryption operation."
```

Reviewer 2: Mm that slightly sounds like nodes have the power to decrypt unilaterally

Reviewer 3: Perhaps. My issue was that "decryption material" could be interpreted as private key material. It's a tough one to word.

```suggestion
    description: "Data requesters present proof to the cohort, and each member verifies condition fulfilment before provisioning the requester with a decryption share; decryption requires a threshold of these shares."
```

    image: "/images/third-taco-diagram.svg"
    buttons:
      - label: "taco-web"
        url: https://github.com/nucypher/taco-web
        variant: EXTERNAL_OUTLINE

>       - label: "nucypher/main"
>         url: https://github.com/nucypher/nucypher/tree/main

Reviewer 3: ```suggestion
      - label: "nucypher"
        url: https://github.com/nucypher/nucypher
```

        variant: EXTERNAL_OUTLINE
tacoExamplesInfo:

>   title: "Access conditions that fit your use case"

Reviewer 3: ```suggestion
  title: "Access Conditions for your Use Case"
```

Reviewer 2: I think that reduces the potency of this title, in saying that you can customize conditions to precisely fit your use case's needs

  preTitle: "build"
  description: "Whatever the business logic for data access, TACo nodes will enforce it. Specify and combine EVM, RPC and time-based conditions. Non-blockchain coming soon."
  image: "/images/ecosystem-grid.svg"
  rowReverse: false
tacoExamples:
  - title: "NFT-gating"

>     description: "Predicate decryption rights on the holding of special-purpose NFTs. Works for digital media, event streams and beyond."

Reviewer 3: ```suggestion
    description: "Predicate decryption rights on ownership of a special-purpose NFTs. Works for digital media, event streams and beyond."
```

    image: "/images/nft-gating.png"
  - title: "Secret recovery"

>     description: "Share your seed phrase or password  – to beneficiaries or your future self – without risking DOS by an intermediary."

Reviewer 3: ```suggestion
    description: "Securely share your seed phrase or password, conditionally with beneficiaries or your future self, without risking DOS by an intermediary."
```

Reviewer 2: ```suggestion
    description: "Securely share your seed phrase/password with beneficiaries or your future self, without risking DOS by an intermediary."
```
'Conditionally' feels a bit clumsy here, but happy to remove the hyphens

    image: "/images/secret-recovery.png"

>   - title: "Inference protection"

Reviewer 3: ```suggestion
  - title: "Inference Protection"
```

Reviewer 2: I generally prefer avoiding capitalizations where possible, hence why 'Secret recovery' etc only has the first word capitalized

Reviewer 3: For smaller sections like this fair. 

Although IMHO for major sections titles like "Access conditions that fit your use case"  (see previous comment) we should keep some measure of capitalization. Either we shorten the statement or make the capitalization work.  Without the capitalization, it becomes inconsistent with other major section titles.

    description: "Prevent third-parties from harvesting or blocking communication between end-users and generative models."

>     image: "/images/inference-protection.png"

Reviewer 3: The abi entry is not correct i.e. the following:
```
functions: 'isAllowedAbi'
```
should instead be (note: no quotes for `isAllowedAbi`) - it's really a variable that was previously defined and not a string.
```
functionAbi: isAllowedAbi,
```

Reviewer 3: Some of the text for the flow chart is blurry.

Reviewer 2: Good spot. 

I already mentioned to @evandrosaturnino re: the flow diagram image quality. Can you change this and the abi entry Derek mentioned?

Author: Addressed in [refactor inference protection svg](https://github.com/threshold-network/website/pull/113/commits/e625f84f66e12e44ec64673f06fb17d6a2fb6bc7)

tacoTableInfo:
  title: "The Zero Compromise Solution"
  highlighted: ["Zero", "Compromise"]

>   description: "Choose a service that’s easy to integrate and that’s actually decentralized."

Reviewer 2: ```suggestion
  description: "Don’t choose between a service that’s easy to integrate and one that’s actually decentralized."
```

Reviewer 4: I don't think I follow with the proposed suggestion. To me it would seem it's hinting that TACo is not easy to integrate.

Reviewer 2: Mmm I thought this clearly implied that TACo doesn't require choosing between these two things, because you can have both at once.

Reviewer 2: How about this? 

```suggestion
  description: "Adding extensions takes weeks. Building a battle-tested decentralized network takes years. Don't accept imitations."
```

Reviewer 3: > Mmm I thought this clearly implied that TACo doesn't require choosing between these two things, because you can have both at once.

Perhaps the balance here with your original suggestion could be:
```suggestion
  description: "Use a service that offers both seamless integration and genuine decentralization, without sacrificing one for the other."
```

Reviewer 2: I like this last suggestion but maybe we can go more succinct: 

```suggestion
  description: "Easy to integrate or genuinely decentralized. No longer a difficult choice."
```

Reviewer 2: We could even make a 'trilemma' diagram here with 'Easy to Integrate', 'Programmable' and 'Decentralized'

Reviewer 2: @evandrosaturnino Let's go with either: 
`Easy to integrate or genuinely decentralized. No longer a difficult choice.`
or 
`Adding extensions takes weeks. Building a battle-tested decentralized network takes years. Don't accept imitations.` 
@derekpierre do you have a preference?

Author: The question is that "Easy to integrate or genuinely decentralized. No longer a difficult choice." implies there is a choice to be made between both, but in fact the best way to put it that TACo is both easy to integrate and decentralized

Reviewer 3: @arjunhassard Admittedly, I have a preference for an explicit statement instead of an implicit statement, leaving the user to interpret it, i.e. something closer to what @evandrosaturnino stated, so something like the above or to blend both:
`TACo combines ease of integration and genuine decentralization. No need to choose.` / 
`TACo blends ease of integration with genuine decentralization. No concession required.`

But if you feel strongly about it, I can be on board with the following:
`Easy to integrate or genuinely decentralized. No longer a choice.`

Reviewer 2: Thanks, I agree with both of you. It's not perfect, but let's go with: 

```suggestion
  description: "TACo blends ease of integration with genuine decentralization. No need to choose between the two."
```

  rowReverse: false
tacoTable:
  headers:
    [
      "Feature / Service Choice",
      "TACo",
      "LIT",
      "EVM Contract",
      "PKI",
      "Cloud Service",
    ]
  features:

>     - label: "Scalable to millions of users"

Reviewer 2: The left hand column text (Scalable to millions of users, Arbitrary access conditons, etc) should be aligned left rather than centered

Author: Addressed in https://github.com/threshold-network/website/pull/113/commits/eb14689ece7a466e8441182e7c6e7206013c3e55
